### ISSUE TITLE
`aria-label` support for `va-link`

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/vets-design-system-documentation#1644]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

When tabbing to a `va-link` that has an `aria-label`, the visible link text is announced by NVDA and the `aria-label` is not announced.

## What I expected to happen

When tabbing to a `va-link` with an `aria-label`, the `aria-label` text is announced by all major screen readers.

## Reproducing
- Formation version:
- Device: Windows 10
- Browser: Chrome + NVDA

Steps to reproduce:

1. On a Windows machine, log into staging with a test user and navigate to [Your appointments](https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/).
2. Activate [NVDA](https://www.nvaccess.org/download/).
3. Press [Tab] to move focus to a Details link.
4. Note that the link is announced as "Details" and not as its aria-label.
5. On a Mac, log into staging with a test user and navigate to [Your appointments](https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/).
6. Activate VoiceOver.
7. Press [Tab] to move focus to a Details link.
8. Note that the link's aria-label is announced as expected.

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [x] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details

This came up in the course of a [staging review for VAOS](https://github.com/department-of-veterans-affairs/va.gov-team/issues/55857).

I think I know why this is happening, but I'd love some DST confirmation! I'm not seeing any [attributes in Storybook](https://design.va.gov/storybook/?path=/docs/components-va-link--default) that map to `aria-label` and I'm not seeing the `aria-label` get passed down to the shadow DOM.

Note that [`va-button` does have a `label` attribute](https://design.va.gov/storybook/?path=/docs/components-va-button--primary) that populates `aria-label` in the shadow DOM. I've tested and confirmed that button `aria-label` text is read as expected.

I believe the big three screen readers all parse web components a little bit differently from one another. Testing with VoiceOver on Mac, I'm hearing the `aria-label` as expected -- so VoiceOver is correctly identifying the `aria-label` on the element. But NVDA is apparently looking for that `aria-label` inside the shadow DOM and not finding it, while ignoring it on the `va-link` component itself. Oddly enough, when I have NVDA just read the full contents of the page straight through, it *does* correctly announce the `aria-label`. But tabbing to the link doesn't work. 🤷 

Note also that [Peter Russo flagged some unusual `va-link` behavior in Slack](https://dsva.slack.com/archives/C8E985R32/p1678395616652319) when testing with Android TalkBack. I don't have an Android device to confirm his findings. But I think there's a decent chance TalkBack is struggling with the same thing NVDA is but in a different way, and both can be resolved by explicitly mapping a component attribute to the `aria-label`.

**Comment by humancompanion-usds** [comment]
This component went through CollabCycle and I believe we had an a11y specialist on the team at the time testing our components. Thus we potentially missed this twice? Third time is a charm? @briandeconinck - Would you mind looking back to see if this was flagged then at all? I'm curious if we decided to not do this for some reason or if it was just missed.

**Comment by rsmithadhoc** [comment]
> @rsmithadhoc - Can you and @briandeconinck get together and discuss whether this is a blocker for folks are not? Sitewide has it as a blocker.

Hi @humancompanion-usds,

I talked to Brian, and we do not believe it is a blocker for teams, as there are workarounds.

- If the link text can be made more descriptive, that’s a workaround (or actually preferred in most scenarios) over adding an `aria-label` to a link.
- I believe Sitewide is saying it is a blocker because in the transition to v3, it wouldn't function exactly as it does now. They have links like this: `<a aria-label="Edit zip code" href="#" name="zipCode">Edit</a>` containing the additional context in an `aria-label`, which `va-link` doesn't currently support. They could update the text to be more descriptive or make the user rely on page context (less preferred) until we add this, which would allow them to use `va-link`.
![image](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/f0922e67-eacb-4c8e-bd0a-bc463b5befc9)

That being said, this update won't be too difficult to add and would be a good addition to get in place.

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/1644?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc
